coverage/PENDING_SUBMIT/amdvlk/0612-COVERAGE-instruction-simplify-2340/cov-conditions-empty-blocks-index-array-one-divided-by-findlsb.amber

```diff
@@ -0,0 +1,285 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific LLVM code paths
+
+# The test passes because the shader always writes red.
+
+# Optimized using spirv-opt with the following arguments:
+# '-O'
+# spirv-opt commit hash: a0370efd589be33d5d9a85cfde2f85841b3755af
+
+
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# #define _int_0 _GLF_uniform_int_values[0]
+# #define _int_16 _GLF_uniform_int_values[1]
+# #define _int_1 _GLF_uniform_int_values[2]
+# #define _float_2_0 _GLF_uniform_float_values[0]
+# #define _float_1_0 _GLF_uniform_float_values[1]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_int_values: [0, 16, 1]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int _GLF_uniform_int_values[3];
+# };
+#
+# // Contents of _GLF_uniform_float_values: [2.0, 1.0]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     float _GLF_uniform_float_values[2];
+# };
+#
+# // Contents of injectionSwitch: [0.0, 1.0]
+# layout(set = 0, binding = 2) uniform buf2
+# {
+#     highp vec2 injectionSwitch;
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     float arr[10] = float[10](_float_1_0, _float_1_0, _float_1_0, _float_1_0, _float_1_0, _float_1_0, _float_1_0, _float_1_0, _float_1_0, _float_1_0);
+#
+#     // Always false.
+#     if(injectionSwitch.x > injectionSwitch.y)
+#     {
+#     }
+#     else
+#     {
+#         // Always false.
+#         if(gl_FragCoord.x < 0.0)
+#         {
+#         }
+#         else
+#         {
+#             // Always false.
+#             if(injectionSwitch.x > injectionSwitch.y)
+#             {
+#             }
+#             else
+#             {
+#                 // arr[0] becomes 2.0.
+#                 arr[_int_1 / findLSB(_int_16)] = _float_2_0;
```

Comment:
Could we change this to:

arr[_int_2 / findLSB(_int_2)]

assuming this sets arr[1] to 2.0, and then we check arr[_int_1] below? And maybe check that elements 0 and 2 are still 0.0? That way, if findLSB(_int_2) is somehow miscalculated, the test will fail.

Reply:
Did the changes, although it needs to be 2 / findlsb(4).